plugins/org.jkiss.dbeaver.ui.navigator/src/org/jkiss/dbeaver/ui/actions/datasource/SelectActiveSchemaHandler.java

```diff
@@ -183,11 +183,17 @@ public void propertyChanged(Object source, int propId) {
                 if (contextDefaults != null) {
                     DBSCatalog defaultCatalog = contextDefaults.getDefaultCatalog();
                     DBSSchema defaultSchema = contextDefaults.getDefaultSchema();
+                    String databaseNameFromConnectionConfiguration = executionContext.getDataSource().getContainer()
```

Comment:
Looks like conceptual behavior change.
Why take database name from configuration if it differs from actual active catalog?